src/python/WMCore/MicroService/MSPileup/MSPileupMonitoring.py

```diff
@@ -0,0 +1,185 @@
+"""
+File       : MSPileupMonitoring.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupMonitoring provides bridge between MSPileup
+             service and CMS Monitoring infrastructure
+
+Usage:
+    # initialize monitoring manager
+    monMgr = MSPileupMonitoring()
+    # start monitoring trhead
+    self.monMgr.startMonThread()
+    # check monitoring thread status
+    status = self.monMgr.isAlive()
+    # restart monitoring thread
+    self.monMgr.restart()
+"""
+
+# system modules
+import time
+import datetime
+
+# WMCore modules
+from WMCore.MicroService.MSCore.TaskManager import start_new_thread
+from WMCore.MicroService.MSCore.MSManager import daemonOpt
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+
+# CMSMonitoring modules
+from CMSMonitoring.StompAMQ7 import StompAMQ7 as StompAMQ
+
+
+# THR_FLAT controls kill process of daemon
+THR_FLAG = False
+
+
+def flatDocuments(doc):
+    "Helper functino to flat out MSPileup document"
+    docs = flatKey(doc, 'ruleIds')
+    docs = (f for d in docs for f in flatKey(d, 'currentRSEs'))
+    docs = (f for d in docs for f in flatKey(d, 'expectedRSEs'))
+    docs = (f for d in docs for f in flatKey(d, 'campaigns'))
+    for doc in docs:
+        yield doc
+
+
+def flatKey(doc, key):
+    "Helper function to flat out values of given key in a document"
+    for item in doc[key]:
+        ndoc = dict(doc)
+        nkey = key[:-1]  # ruleIds -> ruleId
+        ndoc[nkey] = item
+        del ndoc[key]
+        yield ndoc
+
+
+def daemon(func, interval, logger, *args, **kwargs):
```

Comment:
I guess I need to have another look at this logic then. But in general, relying on a global variable isn't a good start. Sorry, I cannot suggest any other alternative at the moment.

Reply:
I agree about globals and I also don't like them. The only alternative I see is to use signal, something like this:
```
def signal_handler(sig, frame):
    print('You pressed Ctrl+C!')

signal.signal(signal.SIGINT, signal_handler)
```
I need to experiment with this but it may allow to replace global with signal handler. Do you want me to try it out?